anesthetic/read/chain.py

```diff
@@ -0,0 +1,29 @@
+"""Read MCMCSamples or NestedSamples from any chains."""
+from anesthetic.read.polychord import read_polychord
+from anesthetic.read.getdist import read_getdist
+from anesthetic.read.cobaya import read_cobaya
+from anesthetic.read.multinest import read_multinest
+
+
+def read_chains(root, *args, **kwargs):
+    """Auto-detect chain type and read from file.
+
+    Parameters
+    ----------
+    root: str
+        root name
+
+    Returns
+    -------
+    NestedSamples or MCMCSamples depending on auto-detection
```

Comment:
I've added burn in to 74fe4135bfe9276763d68755ac113f8acb80986d, as well as a note that all other args and kwargs are forwarded onto `NestedSamples` or `MCMCSamples` constructors. `burn_in` is the only argument specific to reading at the moment (note that `MCMCSamples` no longer has it now we've moved reading out of the constructor).

Reply:
> (note that `MCMCSamples` no longer has it now we've moved reading out of the constructor)

In the long run it might be nice to have some sort of `remove_burn_in` method, that correctly handles the `'chain'` key to remove from all chains, which slicing cannot do.

Reply:
Definitely! Added as a reminder in #219